### PR TITLE
submit benchmark results json to a codespeed instance

### PR DESCRIPTION
**Pull request by dt**
Note: depends on #75, or manually changing the installed version on the hosts that run this job.

I still need to figure out how to test this, but wanted to get a draft out for feedback.

### CODE REVIEW

File: scripts/run_benchmarks.sh [hunk 1]
```diff
@@ -38,13 +38,16 @@ LOGS_DIR="${1-$(mktemp -d)}"
 MAILTO="${MAILTO-}"
 KEY_NAME="${KEY_NAME-cockroach-${USER}}"
 
-SSH_KEY="~/.ssh/${KEY_NAME}.pem"
+SSH_KEY="$HOME/.ssh/${KEY_NAME}.pem"
```
Comment (dt) [inline scripts/run_benchmarks.sh]: I had to switch this since it wasn't expanding for me in the `scp` invocation below.

**Comment by tschottdorf** [comment]
LGTM, but let @mberhault take a look.

File: scripts/run_benchmarks.sh [hunk 2]
```diff
@@ -38,13 +38,16 @@ LOGS_DIR="${1-$(mktemp -d)}"
...
 SSH_USER="ubuntu"
 BINARY_PATH="cockroach/static-tests.tar.gz"
 
 PROD_REPO="${COCKROACH_BASE}/cockroach-prod"
 run_timestamp=$(date  +"%Y-%m-%d-%H:%M:%S")
 
+CODESPEED_SERVER="http://104.196.51.203:8000"
```
Comment (mberhault) [inline scripts/run_benchmarks.sh]: the address should probably be an argument. we don't want to expose that.

File: scripts/upload_benchmarks.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python2.7
```
Comment (mberhault) [inline scripts/upload_benchmarks.py]: any reason not to make this in go?

Comment (dt) [inline scripts/upload_benchmarks.py]: I was vaguely following the suggested upload script provided `codepseed`, just with my own arg parsing, and theirs was in python. Could rewrite it in go but wasn't sure if it really mattered.

File: scripts/run_benchmarks.sh (hunk 2 above)
Comment (dt) [inline scripts/run_benchmarks.sh]: AFAIK the plan is that the codespeed server to be public so we'll need to come up with an auth solution other than obscurity, but we can hide it for now?

File: scripts/upload_benchmarks.py (hunk 3 above)
Comment (mberhault) [inline scripts/upload_benchmarks.py]: I don't think it's a big deal, but it does add another dev env dependency. It's all basic python, so it should be fine. Let's keep it for now, but I may come back and whine :)

File: scripts/run_benchmarks.sh (hunk 2 above)
Comment (mberhault) [inline scripts/run_benchmarks.sh]: I'd rather keep it external for now. I need tiny wrapper scripts around my nightly stuff anyway to specify really custom things (like email address and github token) while others are loaded from file (ssh key, aws credentials)

**Comment by mberhault** [comment]
LGTM, but let's move the server addr to an arg and do something similar to the MAILTO. eg: if it's not set, skip the upload.

File: scripts/upload_benchmarks.py (hunk 3 above)
Comment (dt) [inline scripts/upload_benchmarks.py]: yeah, I expressly avoided any non-stdlib (as of 2.7) deps.

**Comment by dt** [comment]
k, removed `CODESPEED_SERVER` default for now, PTAL

**Pull request merged**